app/helpers/samples_helper.rb

```diff
@@ -345,4 +366,15 @@ def get_ready_sample_ids(samples)
 
     ready_sample_ids
   end
+
+  def get_distinct_sample_types(samples)
+    types = Metadatum
+            .where(key: "sample_type")
+            .where(sample_id: samples.pluck(:id))
+            .map(&:validated_value)
```

Comment:
would use `pluck(:text_validated_value)` instead of `map` here for performance

Reply:
Fixed, thanks!